### PR TITLE
Add ScipyMilpOptimizer

### PR DESCRIPTION
**Pull request by t-imamichi**
### Summary

Adds `ScipyMilpOptimizer` based on Scipy's MILP solver.
Scipy introduced MILP solver since 1.9.0 (supports Python 3.8 or newer).

supersedes #447 

This PR is a QAMP 22 fall project by @Dran-Z https://github.com/qiskit-advocate/qamp-fall-22/issues/2 

### Details and comments

- [x] add more unit tests
- [x] update reno
- [x] ~~update requirements.txt?~~

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
Since we run coverage along with the other checks under 3.7 and this new code is not supported in that version this would explain the coverage drop.  In Nature I had already bumped the checks to run under 3.8 - since we build/deploy docs under 3.8 and we build a documentation zip as part of the checks. I had bumped it since 3.7 is nearing EOL and I wanted to ensure that what I was seeing in the docs was indeed the same as what we published. This is a simple one line change and could be done in a separate PR to check things out - then once the branch is updated here we would get a better idea of the coverage of this new code.

**Comment by t-imamichi** [comment]
Thank you for your information. I will bump Python version for the check.

File: qiskit_optimization/optionals.py [hunk 1]
```diff
@@ -47,3 +47,11 @@
     name="Matplotlib",
     install="pip install 'qiskit-optimization[matplotlib]'",
 )
+
+HAS_SCIPY_MILP = LazyImportTester(
+    {
+        "scipy.optimize": ("milp",),
+    },
+    name="SciPy MILP",
+    install="pip install 'qiskit-optimization[scipy_milp]'",
```
[the remark this answers is not in the document]
Comment (t-imamichi) [reply qiskit_optimization/optionals.py:56]: Thanks. You are right. `pip install scipy>=1.9.0` is enough.

File: qiskit_optimization/algorithms/scipy_milp_optimizer.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    def __init__(self, disp: bool = False) -> None:
+        """Initializes the ScipyMILPOptimizer.
+
+        Args:
+            disp: Whether to print MILP output or not.
+        """
+        self._disp = disp
+
+    @staticmethod
+    def is_scipy_milp_installed():
```
[the remark this answers is not in the document]
Comment (t-imamichi) [reply qiskit_optimization/algorithms/scipy_milp_optimizer.py:59]: Yes, the optional decorator seems enough. I will remove this method.

File: qiskit_optimization/algorithms/scipy_milp_optimizer.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            problem: The problem to be solved.
+
+        Returns:
+            The result of the optimizer applied to the problem.
+
+        Raises:
+            QiskitOptimizationError: If the problem is incompatible with the optimizer.
+        """
+        from scipy.optimize import milp  # pylint: disable=no-name-in-module
+
```
[the remark this answers is not in the document]
Comment (t-imamichi) [reply qiskit_optimization/algorithms/scipy_milp_optimizer.py:166]: Thanks. I remove the check in `_generate_problem` by using `_verify_compatibility` in `solve`.

File: qiskit_optimization/algorithms/scipy_milp_optimizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (t-imamichi) [reply qiskit_optimization/algorithms/scipy_milp_optimizer.py]: done 650715b

**Review by woodsp-ibm — APPROVED** [review]
LGTM. 100% test coverage of the ScipyMilpOptimzer - nice!

**Review by a-matsuo — APPROVED** [review]
LGTM!

**Pull request merged**